pulp_ansible/app/tasks/collections.py

```diff
@@ -574,7 +574,11 @@ async def _fetch_collection_metadata(self, requirements_entry):
         if requirements_entry.version == "*":
             requirement_version = Requirement.parse("collection")
         else:
-            requirement_version = Requirement.parse(f"collection{requirements_entry.version}")
+            try:
+                Version(requirements_entry.version)
```

Comment:
Is this to trigger a Version ValueError before the Requirement.parse()?

Does Requirement.parse() not raise a useful exception?

I'd add a comment explaining that because it looks unusual.

Reply:
It looks like Version() might raise more useful ValueErrors (ie, with explanation messages) than Requirement.parse().

Would it be worthwhile to split this into two try/except blocks?
- one for the try/except around Version() that could potentially log or expose the msg from the exception
- another for Requirement.parse 

Might get more specific errors that way for future troubleshooting.